### DISCUSSION TITLE
Create option to submit an assignment in Hedy

### DISCUSSION THREAD

**Discussion by OnnoEbbens** [hedyorg/hedy#954] [Website]
Sometimes I want the students to submit an assignment that I can grade. It would be great if we could add this to Hedy.

For me I would like to see the following options:
- A teacher can create an assignment with an explanation and a deadline.
- A student can make the assignment just like they make code in the tabs.
- When the student is finished they can submit their assignment (before the deadline).
- Some options for the teacher to give feedback on the assignment.
- ..

I am curious what you think about this idea and if you have more options that you would like to have.

**Comment by Felienne** [comment]
Hi @OnnoEbbens!

Converted this idea to a discussion because there is a bit more to think about. Maybe @MarleenGilsing and you can brainstorm here about what features teachers would need most?

**Reply by MarleenGilsing** [reply]
It makes sense to make a feature like this. Especially if we want teachers to be able to see their students' grades from the quizzes, than they should also be able to grade their programms in assignments.

**Comment by OnnoEbbens** [comment]
I would also like to have some checks in place for students handing in their assignments. So we can give the students a warning if they want to hand in code that raises an exception.

**Comment by Felienne** [comment]
@TiBiBa Might this be a feature you can also pick up? I can imagine it fits into the class customization (enable option t hand in assignments?)

We have to prevent to go on the path of reimplementing magister, so I want to stay away from things like deadline and points etc, teachers can admin those things elsewhere, but the idea of handing in meaning that program is frozen would be very helpful and I guess not all to hard to build? 

Maybe a very simple implementation could be to go to the list of shared programs and just make a "hand in" button there?

**Reply by TiBiBa** [reply]
Sounds good! I will think about a efficient way to implement this and make a mock-up. Do we want to add the assignment as a adventure tab (maybe with a different color?) or take the last saved program per level?

**edit**
Nevermind, missed the part of a hand-in button on the shared programs page. Will take a look and share a mock-up later this week!

**Reply by Felienne** [reply]
Great! I do like the idea of it being a separate adventure but I think that only makes sense when teachers can write their own adventures (and we said we'd stop ding more features for a while haha, so let's keep it like proposed above for now)

**Reply by TiBiBa** [reply]
I'm still struggling a bit on how we want to implement this. For an assignment a teacher might want to give a title, an explanation and required Hedy level (?) We also want to have an efficient way to let the user know there are assignments for them. Do we want to show this on the "My profile" page? I don't really like that idea as I think this page should be dedicated to personal settings and class related information should be placed elsewhere. Maybe add a menu tab similar to "For teachers"? but then "My class" for students? 

Also we would have to create some overview page for the teacher, similar to the class overview, were they are able to see their assignments, edit and delete them. I propose that we use a similar structure to the class customizations, storing the assignments in a database with a unique class/assignment key combination.

TLDR; I think it would be best to have a more extensive implementation as it would make is easier for us to adjust the UI, responds to the teachers needs and implement possible future features.

**Reply by Felienne** [reply]
I love these ideas @TiBiBa, a "my class" page for kids where they can hand in stuff and see their classes. 

However, I think for Onno's use case this extensive implementation is not necessary (and I also agree with your sentiment of keeping things simple)

A very simple and (for now) satisfying implementation would be the following:

Add a "Freeze" button to already shared programs, so that kids can't continue working on that program. They can then simply hand in the link to that program in Magister or Som or something (almost all schools use a digital system for homework, so we must be wary to reimplement such a thing, especially since in many school it is mandatory to put homework in those systems)

When we have teacher created adventures, they can then simply make an "adventure" that is their assignment and then is start to look already more like what we want.

Does that help?
